app/src/main/java/sensors_in_paradise/sonar/util/UIHelper.kt

```diff
@@ -21,5 +23,25 @@ class UIHelper private constructor() {
             disconnectionAlertDialog.show()
             return disconnectionAlertDialog
         }
+
+        fun getPrimaryColor(context: Context): Int {
+            return if (PreferencesHelper.shouldUseDarkMode(context)) {
+                ContextCompat.getColor(context, R.color.colorPrimaryDark)
+            } else {
+                ContextCompat.getColor(context, R.color.colorPrimary)
+            }
+        }
+
+        fun getSlightBackgroundContrast(context: Context): Int {
+            return ContextCompat.getColor(context, R.color.slightBackgroundContrast)
+        }
+
+        fun getBackroundContrast(context: Context): Int {
+            return if (PreferencesHelper.shouldUseDarkMode(context)) {
+                ContextCompat.getColor(context, R.color.lightGrey)
+            } else {
+                ContextCompat.getColor(context, R.color.darkGrey)
+            }
+        }
```

Comment:
Differentiation between colors based on the theme should not be done in code but use the color resources (there is one `colors.xml` file for light mode and one for dark mode).

Reply:
WENN es SEIEN muss :~/